src/pybind/mgr/cephadm/inventory.py

```diff
@@ -1405,6 +1406,175 @@ def get_scheduled_daemon_action(self, host: str, daemon: str) -> Optional[str]:
         return self.scheduled_daemon_actions.get(host, {}).get(daemon)
 
 
+class NodeProxyCache:
+    def __init__(self, mgr: "CephadmOrchestrator") -> None:
+        self.mgr = mgr
+        self.data: Dict[str, Any] = {}
+        self.oob: Dict[str, Any] = {}
+
+    def load(self) -> None:
+        _oob = self.mgr.get_store('node_proxy/oob', "{}")
+        self.oob = json.loads(_oob)
+
+        for k, v in self.mgr.get_store_prefix(NODE_PROXY_CACHE_PREFIX).items():
+            host = k.split('/')[-1:][0]
+
+            if host not in self.mgr.inventory.keys():
+                # remove entry for host that no longer exists
+                self.mgr.set_store(f"{NODE_PROXY_CACHE_PREFIX}/{host}", None)
+                try:
+                    self.oob.pop(host)
```

Comment:
should we also be removing the host from the `data` dict?

Reply:
> should we also be removing the host from the `data` dict?

I think you are right, yes.